angles/src/angles/__init__.py

```diff
@@ -0,0 +1,179 @@
+#  modification, are permitted provided that the following conditions
+#  are met:
+#
+#   * Redistributions of source code must retain the above copyright
+#     notice, this list of conditions and the following disclaimer.
+#   * Redistributions in binary form must reproduce the above
+#     copyright notice, this list of conditions and the following
+#     disclaimer in the documentation and/or other materials provided
+#     with the distribution.
+#   * Neither the name of the Willow Garage nor the names of its
+#     contributors may be used to endorse or promote products derived
+#     from this software without specific prior written permission.
+#
+#  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+#  "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+#  LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+#  FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+#  COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+#  INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+#  BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES
+#  LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+#  CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+#  LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+#  ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+#  POSSIBILITY OF SUCH DAMAGE.
+#********************************************************************/
+
+from math import fmod, pi, fabs
+
+def normalize_angle_positive(angle):
+    """ Normalizes the angle to be 0 to 2*pi
+        It takes and returns radians. """
+    return fmod(fmod(angle, 2.0*pi) + 2.0*pi, 2.0*pi)
+
+def normalize_angle(angle):
+    """ Normalizes the angle to be -pi to +pi
+        It takes and returns radians."""
+    a = normalize_angle_positive(angle)
+    if a > pi:
+        a -= 2.0 *pi
+    return a
+
+def shortest_angular_distance(from_angle, to_angle):
+    """ Given 2 angles, this returns the shortest angular
+        difference.  The inputs and ouputs are of course radians.
+ 
+        The result would always be -pi <= result <= pi. Adding the result
+        to "from" will always get you an equivelent angle to "to".
+    """
+    return normalize_angle(to_angle-from_angle)
+
+def two_pi_complement(angle):
+    """ returns the angle in [-2*pi, 2*pi]  going the other way along the unit circle.
+        \param angle The angle to which you want to turn in the range [-2*pi, 2*pi]
+            E.g. two_pi_complement(-pi/4) returns 7_pi/4
+                 two_pi_complement(pi/4) returns -7*pi/4
+    """
+    #check input conditions
+    if angle > 2*pi or angle < -2.0*pi:
+        angle = fmod(angle, 2.0*pi)
+    if angle < 0:
+        return 2*pi+angle
+    elif angle > 0:
+        return -2*pi+angle
+
+    return 2*pi
+
+def find_min_max_delta(from_angle, left_limit, right_limit):
```

Comment:
This would be great to prefix with an underscore to signify that it's not public API.